### ISSUE TITLE
Refresh stage metadata automatically in the background to avoid exceptions encountered during file upload in uploadWithoutConnection

### ISSUE THREAD

**Issue by ademfreshbooks** [snowflakedb/snowflake-ingest-java#753]
Snowflake Ingest SDK: 2.1.0

For long running clients in streaming mode, It seems ~ every hour, session expires for StreamingIngestStage (used by FlushService), that raises:

```
Exception encountered during file upload in uploadWithoutConnection
```

Then StreamingIngestStage retries request again, by default 5 times. 

Usually this does the job, however for app observability sake, this is not perfect as this creates false positives.

Also, there does not appear to be any configuration options available to avoid this behaviour, or options to override some of the things.

**Can you please try to add auto refresh for stage metadata to avoid this situation?**

**Comment by sfc-gh-lsembera** [comment]
Thanks for the suggestion, @ademfreshbooks. We do attempt to auto-refresh stage metadata when uploads start failing, but we can improve the situation by not logging the initial exception when metadata was immediately successfully refreshed.

**Comment by sfc-gh-lsembera** [comment]
Yeah, the SDK doesn't log the first upload error, but the library we are using to upload data does. Thanks for reporting the issue, we will post updates here once the issue gets prioritized.

**Comment by sfc-gh-tzhang** [comment]
This should be fixed in 2.1.2 release, please give it a try and report back if needed, thanks!
